### PR TITLE
EmbeddedV3: Support passing in signers

### PR DESCRIPTION
**Pull request by VeniaminC**
## Description
Support passing in custom signers

## Test plan
tested EVM / Solana and USDC locally: 

image
image
image
image
image
image

### CODE REVIEW

File: packages/client/base/src/types/embed/crypto.ts [hunk 1]
```diff
@@ -55,3 +55,20 @@ export type ETHEmbeddedCheckoutSigner = CommonEmbeddedCheckoutSignerProps & {
...
+    | {
+          success: false;
+          errorMessage: string;
+      };
+
+export type EmbeddedCheckoutSigner = {
+    address: string;
+    chain: SignerSupportedBlockchains;
+    handleSignAndSendTransaction(serializedTransaction: string): Promise<TransactionResponse>;
+    handleSwitchChain?(chainId: number): Promise<void>;
```
Comment (pgarciaegido) [inline packages/client/base/src/types/embed/crypto.ts:73]: Lets think about this scenario.

File: packages/common/base/src/blockchain/types/index.ts [hunk 2]
```diff
@@ -23,6 +24,13 @@ export const BlockchainIncludingTestnet = {
 } as const;
 export type BlockchainIncludingTestnet = ObjectValues<typeof BlockchainIncludingTestnet>;
 export const BLOCKCHAINS_INCLUDING_TESTNETS = objectValues(BlockchainIncludingTestnet);
+export const SignerSupportedBlockchains = {
+    SOLANA: "solana",
+    ...EVMBlockchain,
+    ...EVMBlockchainTestnet,
+} as const;
+export type SignerSupportedBlockchains = ObjectValues<typeof SignerSupportedBlockchains>;
+export const SIGNER_SUPPORTED_BLOCKCHAINS = objectValues(SignerSupportedBlockchains);
```
Comment (pgarciaegido) [inline packages/common/base/src/blockchain/types/index.ts:33]: Do we use this?

File: packages/client/ui/react-ui/src/components/embed/v3/crypto/SignerConnectionHandler.tsx
Comment (pgarciaegido) [inline packages/client/ui/react-ui/src/components/embed/v3/crypto/SignerConnectionHandler.tsx:1]: This piece Id prefer @mPaella to revisit

File: packages/client/base/src/types/embed/v3/CrossmintEmbeddedCheckoutV3Props.ts [hunk 3]
```diff
@@ -190,6 +190,7 @@ export type EmbeddedCheckoutV3CryptoPayment = {
     enabled: boolean;
     defaultChain?: BlockchainIncludingTestnet;
     defaultCurrency?: CryptoCurrency;
+    signer?: EmbeddedCheckoutSigner;
```
Comment (mPaella) [inline packages/client/base/src/types/embed/v3/CrossmintEmbeddedCheckoutV3Props.ts:193]: payer

Comment (mPaella) [reply packages/client/base/src/types/embed/v3/CrossmintEmbeddedCheckoutV3Props.ts:193]: see todo below

File: packages/client/base/src/types/embed/crypto.ts [hunk 4]
```diff
@@ -55,3 +55,19 @@ export type ETHEmbeddedCheckoutSigner = CommonEmbeddedCheckoutSignerProps & {
...
+    | {
+          success: false;
+          errorMessage: string;
+      };
+
+export type EmbeddedCheckoutSigner = {
+    address: string;
+    chain: SignerSupportedBlockchains;
+    handleSignAndSendTransaction(serializedTransaction: string): Promise<TransactionResponse>;
+};
```
Comment (mPaella) [inline packages/client/base/src/types/embed/crypto.ts:73]: put this under the v3 folder, not with all the old stuff

File: packages/client/ui/react-ui/src/components/embed/v3/crypto/SignerConnectionHandler.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    useEffect(() => {
+        if (iframeClient == null) {
+            return;
+        }
+
+        const signTransactionListener = iframeClient.on(
+            "crypto:send-transaction",
+            async ({ serializedTransaction }) => {
+                try {
+                    const tx = await props.payment.crypto.signer!.handleSignAndSendTransaction(serializedTransaction);
```
Comment (mPaella) [inline packages/client/ui/react-ui/src/components/embed/v3/crypto/SignerConnectionHandler.tsx:17]: handle errror message return

Comment (mPaella) [inline packages/client/ui/react-ui/src/components/embed/v3/crypto/SignerConnectionHandler.tsx:17]: dont use `!`

i dont think u need to pass in all the props, instead just pass in the signer then u dont need to use `!`

**Review by mPaella — APPROVED** [review]

**Pull request merged**